imports/wasi_snapshot_preview1/fs.go

```diff
@@ -638,16 +638,15 @@ func fdReadOrPread(ctx context.Context, mod api.Module, stack []uint64, isPread
 	}
 
 	var nread uint32
-	for i := uint32(0); i < iovsCount; i++ {
-		iov := iovs + i*8
-		offset, ok := mem.ReadUint32Le(ctx, iov)
-		if !ok {
-			return 0, ErrnoFault
-		}
-		l, ok := mem.ReadUint32Le(ctx, iov+4)
-		if !ok {
-			return 0, ErrnoFault
-		}
+	iovsBuf, ok := mod.Memory().Read(ctx, iovs, iovsCount<<3)
```

Comment:
iovsCount<<3 can be defined as a variable so that we can reuse below?(though I hope Go compiler is wise enough to optimize that)

Reply:
I tried both ways and perf is roughly the same, so will change here and also fdWrite